Source/bmalloc/bmalloc/DebugHeap.cpp

```diff
@@ -42,7 +42,7 @@ DebugHeap* debugHeapCache { nullptr };
     
 DEFINE_STATIC_PER_PROCESS_STORAGE(DebugHeap);
 
-#if BOS(DARWIN)
+#if BENABLE(MALLOC_HEAP_BREAKDOWN) || BOS(DARWIN)
```

Comment:
This is too confusing. What I would do is guard everything with `BENABLE(MALLOC_HEAP_BREAKDOWN)` only, and always enable it in BPlatform.h if not already defined and `BOS(DARWIN)`. I've left another review comment to suggest this. Then you can have just the single `BENABLE(MALLOC_HEAP_BREAKDOWN)` guard.

Reply:
I think Apple folks want this on even when malloc heap breakdown is compiled out. We should make sure we don't regress them, but since we don't support malloc stack logging (as far as I know) on linux, this condition seems correct to me

Reply:
I reverted this change to the original one